core/lib/contract_verifier/src/etherscan/mod.rs

```diff
@@ -0,0 +1,459 @@
+            match result {
+                Ok(_) => break,
+                Err(EtherscanError::VerificationPending) => {
+                    if tokio::time::timeout(
+                        POLL_VERIFICATION_RESULT_INTERVAL,
+                        self.stop_receiver.changed(),
+                    )
+                    .await
+                    .is_ok()
+                    {
+                        return Err(VerifierError::Canceled);
+                    }
+                    if get_status_attempts > MAX_POLL_VERIFICATION_RESULT_ATTEMPTS {
+                        tracing::warn!(
+                            "Get verification status timed out for contract with address {:#?}",
+                            verification_request.req.contract_address
+                        );
+                        return Err(ProcessingError::VerificationStatusPollingTimeout.into());
+                    }
+                    continue;
+                }
+                Err(e) => return Err(e.into()),
+            }
+        }
+
+        Ok(())
+    }
+
+    async fn verify(
+        &mut self,
+        verification_request: VerificationRequest,
+        etherscan_verification_id: Option<String>,
+    ) -> Result<(), VerifierError> {
+        let req = verification_request.req.clone();
+        let is_verified = self
+            .client
+            .is_contract_verified(req.contract_address)
+            .await?;
+
+        if is_verified {
+            tracing::info!(
+                "Contract with address {:#?} is already verified",
+                req.contract_address
+            );
+            return Ok(());
+        }
+        let verification_id = match etherscan_verification_id {
+            Some(id) => Some(id),
+            None => {
+                let request_id = self
+                    .send_verification_request(&verification_request)
+                    .await?;
+                // Wait for the verification to be processed
+                if tokio::time::timeout(
+                    POLL_VERIFICATION_RESULT_INTERVAL,
+                    self.stop_receiver.changed(),
+                )
+                .await
+                .is_ok()
+                {
+                    return Err(VerifierError::Canceled);
+                }
+                request_id
+            }
+        };
+
+        // None verification id means that the contract is already verified.
+        if verification_id.is_some() {
+            self.await_verification(&verification_request, verification_id.unwrap())
+                .await?;
+        }
+
+        Ok(())
+    }
+
+    async fn process_next_request(&mut self) -> anyhow::Result<(), VerifierError> {
+        let mut started_at = Instant::now();
+        let verification_request = self.get_next_request().await?;
+        if verification_request.is_none() {
+            return Ok(());
+        }
+
+        let (verification_request, etherscan_verification_id) = verification_request.unwrap();
```

Comment:
This can be simplified to:

```rust
let Some((verification_request, etherscan_verification_id)) = self.get_next_request().await? else {
    return Ok(());
}
```